lib/OpenQA/Schema/Result/Comments.pm

```diff
@@ -132,9 +132,17 @@ Returns C<build_nr>, C<type> and optionally C<description> if C<$self> is tag,
 e.g. 'tag:0123:important:GM' returns a list of '0123', 'important' and 'GM'.
 =cut
 sub tag ($self) {
-    $self->text
-      =~ /\btag:((?<version>[-.@\d\w]+)-)?(?<build>[-.@\d\w]+):(?<type>[-@\d\w]+)(:(?<description>[-.@\d\w]+))?\b/;
-    return $+{build}, $+{type}, $+{description}, $+{version};
+    my $text = $self->text;
```

Comment:
suggest to add at least example in a unit test. Test-driven development will make it very obvious whether this works or not.

Reply:
http://quasar.suse.cz/group_overview/1?fullscreen=1 .. the test is a different problem, it looks like `post_comment_1001` don't work as excepted with tag ...

Reply:
I don't know what you mean by that. I was suggesting TDD because it would tell me 1) what you are testing 2) where it fails. Are we talking about posting comments? Or displaying comments?

Reply:
So it looks indeed good on http://quasar.suse.cz/group_overview/1?fullscreen=1. I assume the remaining problem is that the test helper function `post_comment_1001` doesn't allow you to create recognized tags at all? If you need any help, feel free to ask e.g. in the chat or here.